src/wmr-middleware.js

```diff
@@ -119,6 +121,7 @@ export default function wmrMiddleware({
 		pendingChanges.clear();
 	}
 	watcher.on('change', filename => {
+		console.log('changed', filename);
```

Comment:
When we get a change we'll have to look at our module-graph and see if the file accepts updates:

```
function bubbleUpdates() {
  const mod = moduleGraph.get(filename);
  if (mod.acceptsUpdates) {
    // delete build cache;
    return sendUpdateSignal()
  } else if (mod.dependents.size) {
    // iterate over dependents and call bubbleUpdates && delete build cache &&
    // set needsUpdate: true on the module in the moduleGraph
    return;
  } else {
    return sendReloadSignal();
  }
  return sendReloadSignal();
}
```

When we get a request for a file that contains `?t=x` we know that we need to check the imports, we traverse the imports with our usual `resolveId`, if we see a module from the moduleGraph that has a truthy value for needsUpdate we need to append `?t=x` to that import to invalidate that chain of imports.